CHANGELOG.md

```diff
@@ -10,6 +10,7 @@
 - Expose `pcommon.NewSliceFromRaw` function (#5679)
 - `loggingexporter`: create the exporter's logger from the service's logger (#5677)
 - Add `otelcol_exporter_queue_capacity` metrics show the collector's exporter queue capacity (#5475)
+- Add the Collector flag `--list-components` to show the available components. (#5709)
```

Comment:
This does not look like a flag to me, but like a subcommand (something like `build-info`, `components` or `show-components`?)

Reply:
Yeh I thought about that, but it was more intrusive, and I didn't want to introduce a new thing that the collector wasn't already doing.

Reply:
I think it's likely we will eventually have subcommands, some other FRs also make sense as subcommands, like #5654 or #5223. But I agree that it's a significant addition, we may want to put this behind a [feature gate](https://github.com/open-telemetry/opentelemetry-collector/tree/main/service/featuregate#usage) to signal that we haven't yet fully figured out how the final design will look like.